extensions/authentication-passport/src/__tests__/acceptance/authentication-with-passport-strategy-oauth2-adapter.acceptance.ts

```diff
@@ -0,0 +1,333 @@
+//  |            |     payload: {'client_id': , 'secret': }                     | ****         |
+//  |            |                                                              |  ^           |
+//  |            |              +---------------+                               |  |           |
+//  |            |              |               | <---------[2]-------------    |  |           |
+//  |            |              | Mock          | LB App redirects browser      |  |           |
+//  |            |              | Authorization | to auth server,payload:       |  |           |
+//  |            |              | Server        | {'client_id':,                |  |           |
+//  |            |              | (mock-oauth2- |     'callback_url': app url } |  Stage 1     |
+//  |            |              | social-app.ts)|                               |  |           |
+//  |            |              |               |----+ auth server redirects    |  |           |
+//  |            |              |               |    | browser to login page,   |  |           |
+//  |            |              |               |  [3] client_id and            |  |           |
+//  |            |              |               |    | callback_url are         |  |           |
+//  |            |              |               |<---+ passed as hidden params  |  |           |
+//  |            |              |               |                               |  |           |
+//  |            | -----[4]---> |               |                               |  v           |
+//  |            |   login with |               | -------[5]------------->      | ****         |
+//  |            |    user name |               | login success, auth server    |  ^           |
+//  |            |    /password |               | redirects browser to callback |  |           |
+//  |            |              |               | url with access_code          |  |           |
+//  |            |              |               |                               |  |           |
+//  |            |              |               | <-------------[6]---------    |  |           |
+//  |            |              |               |  LB app requests access token |  Stage 2     |
+//  |            |              |               |  with access_code             |  |           |
+//  |            |              |               |                               |  |           |
+//  |            |              |               | ---------------[7]--------->  |  |           |
+//  |            |              +---------------+       returns access token    |  |           |
+//  |            |                                                              |  v           |
+//  |            |      <------------------------[8]-----------------------     | ****         |
+//  +------------+        LB App returns to browser the access token            +--------------+
+
```

Comment:
Thank you for the detailed explanation. It might be better to have a separate `README.md` though.